components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediator.java

```diff
@@ -290,8 +290,13 @@ private boolean processRequestMessage(MessageContext synCtx, SynapseLog synLog)
                 handleException("Error creating response OM from cache : " + id, synCtx);
             }
             if (CachingConstants.HTTP_PROTOCOL_TYPE.equals(getProtocolType())) {
-                msgCtx.setProperty(NhttpConstants.HTTP_SC, Integer.parseInt(cachedResponse.getStatusCode()));
-                msgCtx.setProperty(PassThroughConstants.HTTP_SC_DESC, cachedResponse.getStatusReason());
+                if (cachedResponse.getStatusCode() != null) {
```

Comment:
Should we check for isEmpty() as well?

Reply:
I don't think it is necessary because a `NullPointerException` is not thrown because of an empty String

Reply:
Is a status code that is null, an accepted value? If not, we need to find why it became null rather than having a null check

Reply:
It becomes null because the cache mediator collector was located where a call to a backend was not made.